SpiceSharp/Sparse/SparseOutput.cs

```diff
@@ -0,0 +1,246 @@
+                        sb.Append("    ");
+                        for (I = StartCol; I <= StopCol; I++)
+                        {
+                            if (printReordered)
+                                Col = I;
+                            else
+                                Col = PrintOrdToIntColMap[I];
+                            sb.AppendFormat(" {0,9}", matrix.Translation.IntToExtColMap[Col]);
+                        }
+                        sb.Append(Environment.NewLine + Environment.NewLine);
+                    }
+                    else
+                    {
+                        if (printReordered)
+                            sb.Append($"Columns {StartCol} to {StopCol}.{Environment.NewLine}");
+                        else
+                            sb.Append($"Columns {matrix.Translation.IntToExtColMap[PrintOrdToIntColMap[StartCol]]} to {matrix.Translation.IntToExtColMap[PrintOrdToIntColMap[StopCol]]}.{Environment.NewLine}");
+                    }
+                }
+
+                // Print every row ...
+                for (I = 1; I <= Size; I++)
+                {
+                    if (printReordered)
+                        Row = I;
+                    else
+                        Row = PrintOrdToIntRowMap[I];
+
+                    if (header)
+                    {
+                        if (printReordered && !data)
+                            sb.AppendFormat("{0,4}", I);
+                        else
+                            sb.AppendFormat("{0,4}", matrix.Translation.IntToExtRowMap[Row]);
+                        if (!data)
+                            sb.Append(' ');
+                    }
+
+                    // ... in each column of the group
+                    for (J = StartCol; J <= StopCol; J++)
+                    {
+                        if (printReordered)
+                            Col = J;
+                        else
+                            Col = PrintOrdToIntColMap[J];
+
+                        pElement = matrix.FirstInCol[Col];
+                        while (pElement != null && pElement.Row != Row)
+                            pElement = pElement.NextInCol;
+
+                        if (data)
+                            pImagElements[J - StartCol] = pElement;
+
+                        if (pElement != null)
+                        {
+
+                            // Case where element exists
+                            if (data)
+                                sb.AppendFormat("{0,10}", pElement.Value.Real.ToString("G3"));
+                            else
+                                sb.Append('x');
+
+                            // Update status variables
+                            if ((Magnitude = (Math.Abs(pElement.Value.Real) + Math.Abs(pElement.Value.Imag))) > LargestElement)
+                                LargestElement = Magnitude;
+                            if ((Magnitude < SmallestElement) && (Magnitude != 0.0))
```

Comment:
@svenboulanger Is it safe to compare with zero here ?

Reply:
SmallestElement is the smallest nonzero element in the matrix. An element of 0 is relatively common and indicates a matrix element that is not used (at that point in time).